File: src/test/java/co/nlighten/jsontransform/functions/TransformerFunctionLongTest.java
package co.nlighten.jsontransform.functions;

import co.nlighten.jsontransform.BaseTest;
import org.junit.jupiter.api.Test;

import java.math.BigDecimal;

public class TransformerFunctionLongTest extends BaseTest {
    @Test
    void convert() {
        var decimals = "123456789.87654321";
        var longVal = BigDecimal.valueOf(new BigDecimal(decimals).longValue());
        assertTransformation(decimals, "$$long:$", longVal);
        assertTransformation(decimals, "$$long():$", longVal);
        assertTransformation(123456789.87654321, "$$long:$", longVal);
        assertTransformation(null, "$$long:$", null);
    }

    @Test
    void maxLong() {
        var max = Long.MAX_VALUE;
        assertTransformation(max, "$$long:$", fromJson(Long.toString(max)));
    }
}

File: javascript/json-transform/src/__tests__/functions/TransformerFunctionLong.test.ts
import { describe, test } from "vitest";
import { assertTransformation } from "../BaseTransformationTest";
import { BigDecimal } from "../../functions/common/FunctionHelpers";
import BigNumber from "bignumber.js";

describe("TransformerFunctionLong", () => {
  test("convert", async () => {
    const decimals = "123456789.87654321";
    const longVal = BigInt(new BigDecimal(decimals).toFixed(0, BigNumber.ROUND_DOWN));
    await assertTransformation(decimals, "$$long:$", longVal);
    await assertTransformation(decimals, "$$long():$", longVal);
    await assertTransformation(123456789.87654321, "$$long:$", longVal);
    await assertTransformation(null, "$$long:$", null);
  });
  test("maxLong", async () => {
    const max = BigInt("9223372036854775807"); // MAX_LONG
    await assertTransformation(max, "$$long:$", max);
  });
});
